sim/src/make/activity_model.rs

```diff
@@ -2,114 +2,264 @@ use crate::{
-                residences,
-                |(home, mut forked_rng)| robot_person(home, &workplaces, map, &mut forked_rng),
+        let residents_cap = residents.len();
+        let workers_cap = workers.len();
+
+        // this saturation figure is an arbitrary guess - we assume that the number of trips will
+        // scale as some factor of the people living and/or working on the map. A number of more
+        // than 1.0 will primarily affect the number of "pass through" trips - people who neither
+        // work nor live in the neighborhood.
+        let trip_saturation = 1.2;
+        let num_trips = (trip_saturation * (residents_cap + workers_cap) as f64) as usize;
+
+        // bound probabilities to ensure we're getting some diveristy of agents
+        let lower_bound_prob = 0.05;
+        let upper_bound_prob = 0.90;
+        let prob_local_resident = if workers_cap == 0 {
+            lower_bound_prob
+        } else {
+            f64::min(
+                upper_bound_prob,
+                f64::max(lower_bound_prob, residents_cap as f64 / num_trips as f64),
             )
-            .into_iter()
-            .flatten()
-        {
-            person.id = PersonID(s.people.len());
-            s.people.push(person);
-        }
+        };
+        let prob_local_worker = f64::min(
+            upper_bound_prob,
+            f64::max(lower_bound_prob, workers_cap as f64 / num_trips as f64),
+        );
 
-        // Create trips between map borders. For now, scale the number by the number of residences.
-        let incoming_connections = map.all_incoming_borders();
-        let outgoing_connections = map.all_outgoing_borders();
-        for mut person in timer
+        info!(
+            "BUILDINGS - workplaces: {}, residences: {}, mixed: {}",
+            prettyprint_usize(num_bldg_commercial),
+            prettyprint_usize(num_bldg_residential),
+            prettyprint_usize(num_bldg_mixed_residential_commercial)
+        );
+        info!(
+            "CAPACITY - workers_cap: {}, residents_cap: {}, prob_local_worker: {:.1}%, \
+             prob_local_resident: {:.1}%",
+            prettyprint_usize(workers_cap),
+            prettyprint_usize(residents_cap),
+            prob_local_worker * 100.,
+            prob_local_resident * 100.
+        );
+
+        let mut num_trips_local = 0;
+        let mut num_trips_commuting_in = 0;
+        let mut num_trips_commuting_out = 0;
+        let mut num_trips_passthru = 0;
+        timer.start("create people");
+
+        let all_outgoing_borders = map.all_outgoing_borders();
+        let mut tmp_rng = abstutil::fork_rng(rng);
+        let mut get_commuter_border = || {
+            for _ in 0..50 {
+                // TODO: prefer larger thoroughfares to better reflect reality.
+                let border = all_outgoing_borders.choose(&mut tmp_rng).unwrap();
+
+                // Only consider two-way intersections, so the agent can return the same way
```

Comment:
Could be worth creating this list of bidi borders once outside the loop and sampling from it, for simplicity / marginal perf boost. But if you're planning to add the TODO behavior later, then that makes less sense

Reply:
It's fine for now. Somebody would need to very carefully draw a map boundary to avoid all two-way borders. It won't happen in any of the main "supported" maps right now.